PLMS/Controllers/TaskController.cs

```diff
@@ -1,115 +1,159 @@
+            if (!ModelState.IsValid)
             {
-                return BadRequest();
+                return ApiResponseHelper.CreateErrorResponse(ModelState);
             }
+            var taskDto = _mapper.Map<EditTaskDto>(model);
 
             try
             {
-                await _taskService.EditTaskAsync(task);
+                string userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+                await _taskService.EditTaskAsync(taskDto, id, userId);
             }
             catch (DbUpdateConcurrencyException)
             {
-                return BadRequest();//ToDo: This is a server error, not the client, therefore return 500 Internal Server Error Response
+                return ApiResponseHelper.CreateResponse(false, "DbUpdateConcurrencyException", 500);
+            }
+            catch (UnauthorizedAccessException ex) 
+            {
+                return Unauthorized(ex.Message);
+            }
+            catch (NotFoundException ex)
+            {
+                return NotFound(ex.Message);
             }
 
-            return NoContent();
+            return ApiResponseHelper.CreateResponse(true, "Task was updated successfully");
         }
 
         [HttpPost]
-        public async Task<ActionResult<TaskDto>> AddAsync(TaskDto task)//ToDo: should not accept/return DTO, should accept/return Model/ViewModel instead
+        public async Task<ActionResult<AddTaskModel>> AddAsync(AddTaskModel model)
         {
-            //ToDo: validation for write operations is a must, so implement it, for example, with fluent validator
-            await _taskService.AddTaskAsync(task);
+            string userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            if (!ModelState.IsValid)
+            {
+                return ApiResponseHelper.CreateErrorResponse(ModelState);
+            }
+            var taskDto = _mapper.Map<AddTaskDto>(model);
+            await _taskService.AddTaskAsync(taskDto, userId);
+            var createdTaskModel = _mapper.Map<AddTaskModel>(taskDto);
 
-            return CreatedAtAction(nameof(GetByIdAsync), new { id = task.Id }, task);
+            return ApiResponseHelper.CreateResponse(true, "Task was added successfully");
+            // CreatedAtAction(nameof(GetByIdAsync), new { id = taskDto.Id }, createdTaskModel);  -- Не получается вернуть маршрут на новое айди, не могу понять в чем ошибка (System.InvalidOperationException: No route matches the supplied values.
+         //   at Microsoft.AspNetCore.Mvc.CreatedAtActionResult.OnFormatting(ActionContext context)
         }
 
         [HttpDelete]
         public async Task<ActionResult> DeleteAsync(int id)
         {
-            var task = await _taskService.GetTaskByIdAsync(id);
+            string userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            GetTaskDto taskDto;
+            try
+            {
+                taskDto = await _taskService.GetTaskByIdAsync(id, userId);
```

Comment:
Getting task here is redundant as the TaskService.DeleteTaskAsync is already idempotent, you can remove it